### PR TITLE
Custom key

### PR DESCRIPTION
**Pull request by EliasEriksson**
implemented a issuefab specific key.

### CODE REVIEW

File: Key/Creatable.spec.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+				},
+				a1b2c3d4: {
+					org: true,
+					payment: { expense: true },
+					delegation: true,
+				},
+			},
+		}
+		expect(issuefab.Key.Creatable.is(key)).toEqual(true)
+		expect(issuefab.Key.Creatable.type.get(key)).toEqual(key)
```
Comment (TheLillyleaf) [inline Key/Creatable.spec.ts:25]: Add random wrong object to test getter

File: Key/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		expires: "2023-08-19T00:25:25.000Z",
+		issued: "2023-08-18T12:08:21.000Z",
+		issuer: "userwidgets",
+		token:
+			"eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ1c2Vyd2lkZ2V0cyIsImlhdCI6MTY5MjM2MTUyNSwiYXVkIjoiaXNzdWVmYWIiLCJleHAiOjE2OTI0MDQ3MjUsIm5hbSI6eyJmaXJzdCI6Implc3NpZSIsImxhc3QiOiJkb2UifSwic3ViIjoiamVzc2llQGV4YW1wbGUuY29tIiwicGVyIjoiKi5vcmcgKi5wYXltZW50LmV4cGVuc2UgKi5kZWxlZ2F0aW9uICouYXBwLnZpZXcgYTFiMmMzZDQub3JnIGExYjJjM2Q0LnBheW1lbnQuZXhwZW5zZSBhMWIyYzNkNC5kZWxlZ2F0aW9uIn0.irrOnQOKWlG5TcD1PxBCbiVFQYhMVbueKvtvFyBIy8ueEGftgvltCNhBfW3cx0AJunSP-f6rQwjPFU4huWQlo54UDOTXk7AX6igLDSThB44W-eRCUGHclid_67KIwsq7c5uVqWAbb4UMLILWHMg8ooYafKpeCtjlfLNNRTlYHduZz1dLQSoPqJR40Jpl0UmfKMGXXP-OXQH9cStKg2ZY9L76prhFcH1wTcCFp4-1GjpxI8Ewd-O_djpkRJ8OJCq07X7BckJAyOOo2dV5dOZW9Bb-0ncfKKNDmY2qnhw3jHWO8R5HF0yvkUF05edh_7uE57dPM3vqpspHkfoU_VvORQ",
+	}
+	it("is", () => {
+		expect(issuefab.Key.is(key)).toEqual(true)
+		expect(issuefab.Key.Creatable.type.get(key)).not.toEqual(undefined)
+		expect(issuefab.Key.type.get(key)).toEqual(key)
```
Comment (TheLillyleaf) [inline Key/index.spec.ts:74]: Same here.

File: Key/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		if (token == undefined) {
+			expect(token).not.toEqual(undefined)
+			return
+		}
+		const verifier = issuefab.Key.Verifier.create(publicKey)
+		const key = await verifier.verify(token)
+		if (key == undefined) {
+			expect(key).not.toEqual(undefined)
+			return
+		}
```
Comment (TheLillyleaf) [inline Key/index.spec.ts:88]: How does this work ?

**Review by TheLillyleaf — CHANGES_REQUESTED** [review]
Add check to two tests

File: Key/index.spec.ts (hunk 3 above)
Comment (EliasEriksson) [reply Key/index.spec.ts:90]: key is either undefined or a key. if its undefined the test should fail which is fixed with the expect (no if is necessary for that) but typescript will not understand that jest will stop executing after the failed expect so if key is undefined we return making typescript understand that after the if key is guaranteed to be a key and not undefined. 

So with this setup jest will error and typescript understand that execution will stop if that happens.

**Review by TheLillyleaf — APPROVED** [review]

**Pull request merged**